src/providers/wfs/qgswfsdataitems.cpp

```diff
@@ -123,13 +124,31 @@ QVector<QgsDataItem *> QgsWfsConnectionItem::createChildren()
     QVector<QgsDataItem *> layers;
     if ( capabilities.errorCode() == QgsWfsCapabilities::NoError )
     {
+      const QString projectCrs = QgsProject::instance()->crs().authid();
       const auto featureTypes = capabilities.capabilities().featureTypes;
       for ( const QgsWfsCapabilities::FeatureType &featureType : featureTypes )
       {
+        // if project CRS is supported then use it, otherwise use first available CRS (which is the default CRS)
+        QString crs;
+        if ( !featureType.crslist.isEmpty() )
+        {
+          for ( const QString &c : std::as_const( featureType.crslist ) )
+          {
+            if ( c.compare( projectCrs, Qt::CaseInsensitive ) )
```

Comment:
I assume this should be...

```suggestion
            if ( c.compare( projectCrs, Qt::CaseInsensitive ) == 0 )
```

Reply:
Oops, you are right. Fixed.